packages/xen-api/src/index.js

```diff
@@ -1004,6 +1008,53 @@ export class Xapi extends EventEmitter {
       delete taskWatchers[ref]
     }
   }
+  async _findCurrentMaster() {
+    debug('%s: will try to find a new master', this._humanId)
+    const hosts = await this.getAllRecords('host')
```

Comment:
How can we e call this method if we are trying to open a new session?

Reply:
if the master is down,   EHOSTUNREACH . We wil then try to connect to the fall back hosts